test/apps/iodemo/io_demo.cc

```diff
@@ -614,15 +798,15 @@ class DemoClient : public P2pDemoCommon {
                 break;
             }
 
-            size_t conn_num = IoDemoRandom::rand(0, conn.size() - 1);
+            size_t conn_num = IoDemoRandom::rand(size_t(0), conn.size() - 1);
             io_op_t op      = get_op();
             size_t size;
             switch (op) {
             case IO_READ:
-                size = do_io_read(conn[conn_num], total_iter);
+                size = do_io_read(conn[conn_num], total_iter + sn_shift);
```

Comment:
maybe have separate counting var for "sn"

Reply:
inside of `do_io_read` it becomes "sn", out of the func IMO iteration [0; N] looks looks more clear

Reply:
IMO it's more clear we have sn which may be independent of counting iterations, we have extra var anyway